docs/grammar/SolidityLexer.g4

```diff
@@ -220,6 +220,12 @@ fragment EvenHexDigits: HexCharacter HexCharacter ('_'? HexCharacter HexCharacte
 //@doc:inline
 fragment HexCharacter: [0-9A-Fa-f];
 
+/**
+ * Scanned but not used by any rule, i.e, disallowed.
+ */
+OctalNumber: '0' DecimalDigits ('.' DecimalDigits)?;
```

Comment:
Technically, `8` and `9` are not valid octal numbers, but are present in `DecimalDigits` (at least I'd hope, I haven't looked :D). So something like `091` would not be a valid octal number in the first place, but the parser error would indicate that octal numbers are not supported, which could be a bit confusing if someone makes a typo, e.g. `091` instead of `91`. I think we might have to introduce `OctalDigits` as well if we want to be fully correct, at least I'm assuming that's something @ekpyron or @cameel would say.

Reply:
I think we can leave it as is because solc's parser behaves the same way. I.e. `091` results in
```
Error: Octal numbers not allowed.
```

Perhaps we should change that in solc, but that's a separate matter - and it would still be an error, just a different one (what is even `091`?).

In any case, we should definitely have a case for something like `091` to make sure both the grammar and the parser at least both treat it as an error.

Reply:
We could at least add a note about this weirdness in the docstring above.

Reply:
I was mainly focused in mirroring `solc` parser, which seems to treat anything starting with a `0` (and not followed by `.`) as octal...
I am going to add the test case regarding `091`.

Reply:
Also added the note in the docstring.